python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -618,23 +618,40 @@ def _store_output(
     manager_materializations = []
     manager_metadata: Dict[str, MetadataValue] = {}
 
-    # output_manager.handle_output is either a generator function, or a normal function with or
-    # without a return value. In the case that handle_output is a normal function, we need to
-    # catch errors should they be raised before a return value. We can do this by wrapping
-    # handle_output in a generator so that errors will be caught within iterate_with_context.
+    # TODO - should we handle the op case?
 
-    if not inspect.isgeneratorfunction(output_manager.handle_output):
-
-        def _gen_fn():
-            gen_output = output_manager.handle_output(output_context, output.value)
-            for event in output_context.consume_events():
-                yield event
-            if gen_output:
-                yield gen_output
+    if (
+        output_context.has_asset_key
+        and output_context.dagster_type.is_nothing
+        and output.value is None
+    ):
+        # if the output is from an asset and None is returned, and the type annotation indicates a
+        # Nothing return type don't invoke an I/O manager.
+        def _no_op():
+            yield None
```

Comment:
could we do empty iterator instead to avoid [1] - 
```
def _no_op():
  yield from ()
```
coming from https://stackoverflow.com/questions/13243766/how-to-define-an-empty-generator-function

Reply:
oo that looks much better